source/Firmata/UwpFirmata.cpp

```diff
@@ -234,7 +238,156 @@ UwpFirmata::processInput(
     void
     )
 {
-    return ::RawFirmata.processInput();
+    uint16_t data = _firmata_stream->read();
+    if( data == static_cast<uint16_t>( -1 ) ) return;
+    
+    uint8_t byte = data & 0x00FF;
+    uint8_t upper_nibble = data & 0xF0;
+    uint8_t lower_nibble = data & 0x0F;
+
+    /*
+     * the relevant bits in the command depends on the value of the data byte. If it is less than 0xF0 (start sysex), only the upper nibble identifies the command
+     * while the lower nibble contains additional data
+     */
+    Command command = static_cast<Command>( ( data < static_cast<uint16_t>( Command::START_SYSEX ) ? upper_nibble : byte ) );
+
+    //determine the number of bytes remaining in the message
+    size_t bytes_remaining = 0;
+    bool isMessageSysex = false;
+    switch( command )
+    {
+    default: //command not understood
+    case Command::END_SYSEX: //should never happen
+        return;
+
+        //commands that require 2 additional bytes
+    case Command::DIGITAL_MESSAGE:
+    case Command::ANALOG_MESSAGE:
+    case Command::SET_PIN_MODE:
+    case Command::PROTOCOL_VERSION:
+        bytes_remaining = 2;
+        break;
+
+        //commands that require 1 additional byte
+    case Command::REPORT_ANALOG_PIN:
+    case Command::REPORT_DIGITAL_PIN:
+        bytes_remaining = 1;
+        break;
+
+        //commands that do not require additional bytes
+    case Command::SYSTEM_RESET:
+        //do nothing, as there is nothing to reset
+        return;
+
+    case Command::START_SYSEX:
+        //this is a special case with no set number of bytes remaining
+        isMessageSysex = true;
+        break;
+    }
+
+    //read the remaining message
+    std::vector<uint8_t> vector;
+    size_t bytes_read = 0;
+    while( bytes_remaining || isMessageSysex )
+    {
+        data = _firmata_stream->read();
+        if( data == static_cast<uint16_t>( -1 ) ) continue;
```

Comment:
This puts it into a tight loop if the device suddenly stops sending data.  Should we consider adding a timeout on this?